inbm/cloudadapter-agent/cloudadapter/cloud/client/connections/mqtt_connection.py

```diff
@@ -95,8 +95,10 @@ def _set_proxy(self, config: ProxyConfig) -> None:
         @param config: (ProxyConfig) Config to use
         """
         if config.endpoint:
-            socks.set_default_proxy(socks.PROXY_TYPE_HTTP, *config.endpoint)
-            socket.socket = socks.socksocket  # type: ignore
+            proxy_addr, proxy_port = config.endpoint
+            self._client.proxy_set(proxy_type=socks.HTTP,
+                                   proxy_addr=proxy_addr,
+                                   proxy_port=proxy_port)
```

Comment:
**<h2>$\color{#FFA500}Improvements: CodeQuality$</h2>**
**Suggestion:** Ensure that the proxy_type is dynamically set based on the ProxyConfig instead of hardcoding it to socks.HTTP.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if config.endpoint:
            proxy_addr, proxy_port = config.endpoint
            proxy_type = config.type if hasattr(config, 'type') else socks.HTTP
            self._client.proxy_set(proxy_type=proxy_type,
                                   proxy_addr=proxy_addr,
                                   proxy_port=proxy_port)
```